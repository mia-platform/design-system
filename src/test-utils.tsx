/**
 * Copyright 2023 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { RenderOptions, RenderResult, render } from '@testing-library/react'
import { ConfigProvider } from 'antd'
import { ReactNode } from 'react'

import { ThemeContext, defaultTheme } from './components/ThemeProvider'
import { ANT_PREFIX } from './components/ThemeProvider/Ant'

/**
 * Wraps a React node with a series of providers for testing purposes.
 *
 * @remarks It adds the default theme to components
 * @remarks It adds the default Ant class prefix
 * @remarks It removes the css-do-not-override classes from Ant Design components
 *
 * @param {ReactNode} children - The React node to be wrapped.
 * @returns {ReactNode} The wrapped React node.
 */
const TestWrapper = ({ children }: {children: ReactNode}): ReactNode => (
  <ThemeContext.Provider value={defaultTheme}>
    <ConfigProvider prefixCls={ANT_PREFIX} theme={{ hashed: false }}>
      {children}
    </ConfigProvider>
  </ThemeContext.Provider>
)

/**
 * Custom render function for testing React components.
 *
 * @param {ReactNode} element - The React node to be rendered.
 * @param {RenderOptions} [options] - Options for rendering.
 * @returns {RenderResult} The result of rendering the React node.
 */
const customRender = (
  element: ReactNode,
  options?: Omit<RenderOptions, 'wrapper'>,
): RenderResult => render(element, { wrapper: TestWrapper, ...options })

export * from '@testing-library/react'
export {
  customRender as render,
  TestWrapper,
}
