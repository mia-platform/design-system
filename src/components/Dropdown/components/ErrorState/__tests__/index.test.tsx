/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ErrorState, ErrorStateProps } from '../'
import { RenderResult, fireEvent, render, screen } from '../../../../../test-utils'

describe('ErrorState Component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders with default props', () => {
    const { asFragment } = renderErrorState()

    expect(screen.getByRole('img')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with message', () => {
    const message = 'Error occurred'
    const { asFragment } = render(<ErrorState message={message} />)

    expect(screen.getByRole('img')).toBeInTheDocument()
    expect(screen.getByText(message)).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders with retry button and calls onRetry when clicked', () => {
    const onRetry = jest.fn()
    render(<ErrorState onRetry={onRetry} />)

    const retryButton = screen.getByRole('button', { name: 'Retry' })
    expect(retryButton).toBeInTheDocument()

    fireEvent.click(retryButton)
    expect(onRetry).toHaveBeenCalledTimes(1)
  })

  test('renders with message and retry button', () => {
    const message = 'Error occurred'
    const onRetry = jest.fn()
    render(<ErrorState message={message} onRetry={onRetry} />)

    expect(screen.getByText(message)).toBeInTheDocument()
    expect(screen.getByRole('button', { name: 'Retry' })).toBeInTheDocument()
  })
})

function renderErrorState(props: ErrorStateProps = {}): RenderResult {
  return render(<ErrorState {...props} />)
}
