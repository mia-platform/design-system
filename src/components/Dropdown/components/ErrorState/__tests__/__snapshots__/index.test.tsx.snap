// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorState Component renders with default props 1`] = `
<DocumentFragment>
  <div
    class="error"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM165.66,154.34a8,8,0,0,1-11.32,11.32L128,139.31l-26.34,26.35a8,8,0,0,1-11.32-11.32L116.69,128,90.34,101.66a8,8,0,0,1,11.32-11.32L128,116.69l26.34-26.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`ErrorState Component renders with message 1`] = `
<DocumentFragment>
  <div
    class="error"
  >
    <svg
      color="currentColor"
      fill="currentColor"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: currentColor;"
      viewBox="0 0 256 256"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM165.66,154.34a8,8,0,0,1-11.32,11.32L128,139.31l-26.34,26.35a8,8,0,0,1-11.32-11.32L116.69,128,90.34,101.66a8,8,0,0,1,11.32-11.32L128,116.69l26.34-26.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
      />
    </svg>
    <div
      class="mia-platform-typography bodyS"
      role="paragraph"
    >
      Error occurred
    </div>
  </div>
</DocumentFragment>
`;
